cpp/core/jni/JniWrapper.cc

```diff
@@ -830,7 +830,8 @@ JNIEXPORT jlong JNICALL Java_io_glutenproject_vectorized_ShuffleWriterJniWrapper
     jint startPartitionId,
     jint pushBufferMaxSize,
     jobject partitionPusher,
-    jstring partitionWriterTypeJstr) {
+    jstring partitionWriterTypeJstr,
+    jboolean sortBeforeRR) {
```

Comment:
Would the argument `sortBeforeRR` be used on the native side?

Reply:
Yes. RoundRobinPartitioner implement should know whether drop the first sorted column.

Reply:
Would it be possible to add another projection before to drop the extra column?

Reply:
It's possible, but we can leverage current design to eliminate  the first column and avoid the performance loss by providing extra project.

Reply:
I doubt that the impact of performance loss by adding one project which is only used to drop one column. The most part of extra overhead should come from the sort node. By adding this projection, we can also avoid adding argument `sortBeforeRR` to the several layers in the call stack.

Reply:
Sounds great. I will address this comment later.